imap_processing/spice/geometry.py

```diff
@@ -1 +1,128 @@
-"""Functions for computing geometry using SPICE."""
+"""Functions for computing geometry, many of which use SPICE."""
+
+from collections.abc import Iterable
+from enum import Enum
+from typing import NamedTuple, Optional, Union
+
+import numpy as np
+import spiceypy as spice
+
+from imap_processing.spice.kernels import ensure_spice
+
+
+class SpiceId(NamedTuple):
+    """Class that represents a unique identifier in the NAIF SPICE library."""
+
+    strid: str
+    numid: int
+
+
+class SpiceBody(Enum):
+    """Enum containing SPICE IDs for bodies that we use."""
+
+    # A subset of IMAP Specific bodies as defined in imap_wkcp.tf
```

Comment:
Make certain to put a TODO for change them to positive once IMAP is launched. I believe that's the standard.

Reply:
Not according to the documentation:

> THE SPICE convention uses negative integers as spacecraft ID codes. The code assigned to interplanetary spacecraft is normally the negative of the code assigned to the same spacecraft by JPL's Deep Space Network (DSN) as determined the NASA control authority at Goddard Space Flight Center.

The current SPICE vehicle code assignments:

      NAIF ID     NAME
      ________    ____________________
      -1          'GEOTAIL'
      -3          'MOM'
      -3          'MARS ORBITER MISSION'
        ...
      -41         'MARS EXPRESS'
      -43         'IMAP'
       ...
https://naif.jpl.nasa.gov/pub/naif/toolkit_docs/C/req/naif_ids.html#Spacecraft